roboflow/core/version.py

```diff
@@ -85,54 +89,77 @@ def __init__(self, version_dict, type, api_key, name, version, model_format, loc
                 self.model = None
 
     def download(self, model_format=None, location=None):
+        """
+        Download and extract a ZIP of a version's dataset in a given format
+
+        :param model_format: A format to use for downloading
+        :param location: An optional path for saving the file
 
+        :return: Dataset
+        """
         if location is None:
-            if "DATASET_DIRECTORY" in os.environ:
-                location = (
-                    os.environ["DATASET_DIRECTORY"]
-                    + "/"
-                    + self.name.replace(" ", "-")
-                    + "-"
-                    + self.version
-                )
-            else:
-                location = self.name.replace(" ", "-") + "-" + self.version
+            location = self.__get_download_location()
 
         if not os.path.exists(location):
             os.makedirs(location)
 
-        if model_format is None:
-            if self.model_format == "yolov5":
-                model_format = "yolov5pytorch"
-            elif self.model_format == "yolov7":
-                model_format = "yolov7pytorch"
-            else:
-                RuntimeError(
-                    "You must pass a download_type to version.download() or define model in your Roboflow object"
-                )
-
-        if model_format == "yolov5":
-            model_format = "yolov5pytorch"
-
-        if model_format == "yolov7":
-            model_format = "yolov7pytorch"
+        model_format = self.__get_format_identifier(model_format)
```

Comment:
i tried to simplify/DRY up the `model_format` logic above and also fixed the issue in that the `RuntimeError` was never being `raised`. i believe that my added unit tests should cover all of the cases.